### PR TITLE
Rewrite DelegatingProvider to remove deprecated SortedList

### PR DESCRIPTION
**Pull request by erik-zicht**

### CODE REVIEW

**Comment by erik-zicht** [comment]
retest this please

File: src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php [hunk 1]
```diff
@@ -37,15 +37,18 @@ public function __construct()
      */
     public function addProvider(Provider $provider, $priority = 0)
     {
-        $this->providers->insert($provider, $priority);
+        $this->providers[] = [
+            'priority' => $priority,
+            'provider' => $provider
+        ];
```
Comment (boudewijn-zicht) [inline src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php]: Why not sort when we add a provider?  I.e. add `$this->providers = iterable($this->providers)->sorted('priority')->values()`, this should provide a performance benefit.

File: tests/ZichtTest/Bundle/UrlBundle/Url/DelegatingProviderTest.php [hunk 2]
```diff
@@ -12,10 +12,16 @@ public function __construct($suggestions)
             $this->suggestions = $suggestions;
         }
 
-        public function supports($object) {}
-        public function url($object, array $options = array()) {}
+        public function supports($object)
+        {
+        }
+
+        public function url($object, array $options = array())
```
Comment (boudewijn-zicht) [inline tests/ZichtTest/Bundle/UrlBundle/Url/DelegatingProviderTest.php]: Array short notation

File: src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php [hunk 3]
```diff
@@ -87,16 +90,24 @@ public function suggest($pattern)
...
         return $ret;
     }
+
+    /**
+     * @return Provider[]|array
+     */
+    private function getProviders()
+    {
+        return iterable($this->providers)->sorted('priority')->map('provider')->values();
+    }
```
Comment (boudewijn-zicht) [inline src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php]: When we sort during the `addProvider`, then we will not need this anymore, we can just iterate over `$this->providers` again

**Review by boudewijn-zicht — CHANGES_REQUESTED** [review]

File: src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php (hunk 1 above)
Comment (erik-zicht) [reply src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php]: Is this a performance benefit when you add, lets say, 20 providers and don't use this class at all? Providers are added runtime with every request I assume

**Comment by erik-zicht** [comment]
Commit is now part of #34

**Pull request closed without merging**